### PR TITLE
TaskDefinitionCachingAgent no longer makes describe calls for already cached task definitions.

### PR DESCRIPTION
**Pull request by dkirillov**
@BrunoCarrier

### CODE REVIEW

**Comment by dkirillov** [comment]
image

Red/black deploy still works.

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskDefinitionCachingAgent.java [hunk 1]
```diff
@@ -75,18 +92,45 @@ public String getAgentType() {
...
       ListTaskDefinitionsResult listTaskDefinitionsResult = ecs.listTaskDefinitions(listTasksRequest);
       List<String> taskDefinitionArns = listTaskDefinitionsResult.getTaskDefinitionArns();
+
       if (taskDefinitionArns.size() == 0) {
         continue;
       }
-      for (String taskDefintionArn : taskDefinitionArns) {
-        TaskDefinition taskDefinition = ecs.describeTaskDefinition(new DescribeTaskDefinitionRequest().withTaskDefinition(taskDefintionArn)).getTaskDefinition();
+
+      Set<String> newTaskDefArns = new HashSet<>(taskDefinitionArns);
```
Comment (BrunoCarrier) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskDefinitionCachingAgent.java]: newTaskDefArns could be renamed to uncachedTaskDefArns, and modify the right-hand side to something like:

`new HashSet<>(new HashSet(taskDefinitionArns).removeAll(newTaskDefArns))`;

**Comment by BrunoCarrier** [comment]
+2 LGTM

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskDefinitionCachingAgent.java (hunk 1 above)
Comment (dkirillov) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskDefinitionCachingAgent.java]: Talked it over - no changes required.

**Pull request merged**